java-util/src/main/java/org/apache/druid/java/util/http/client/NettyHttpClient.java

```diff
@@ -215,6 +216,9 @@ public void messageReceived(ChannelHandlerContext ctx, MessageEvent e)
                       log.debug("[%s] Resumed reads from channel (chunkNum = %,d).", requestDesc, resumeChunkNum);
                     }
                   }
+
+                  long backPressureDuration = System.nanoTime() - backPressureStartTimeNs;
```

Comment:
I think backpressure duration should only be computed like this if we're in the `suspendWatermark >= 0 && resumeWatermark >= suspendWatermark` block above. Two reasons:

1. Access to backPressureStartTimeNs must be synchronized, since it's being accessed from multiple threads and it's not a volatile or atomic.
2. If we didn't actually resume anything, then we don't know if backpressure was actually happening (maybe, maybe not) so we shouldn't report a time. In this case, we should return 0.

Reply:
Thanks for the review @gianm! Have made the requested change.